src/audio/codec_adapter/codec/cadence.c

```diff
@@ -444,7 +444,26 @@ int cadence_codec_prepare(struct comp_dev *dev)
 			 ret);
 		goto free;
 	}
-
+	/* Check init done status. Note, it may happen that init_done flag will return
+	 * false value, this is normal since some codec variants needs input in order to
+	 * fully finish initialization. That's why at codec_adapter_copy() we call
+	 * codec_init_process() base on result obtained below.
+	 */
+#ifdef CONFIG_CADENCE_CODEC_WRAPPER
+	/* TODO: remove the "#ifdef CONFIG_CADENCE_CODEC_WRAPPER" once cadence fixes the bug
+	 * in the init/prepare sequence. Basically below API_CALL shall return 1 for
+	 * PCM streams and 0 for compress ones. As it turns out currently it returns 1
+	 * in both cases so in turn compress stream won't finish its prepare during first copy
+	 * in codec_adapter_copy().
+	 */
+	API_CALL(cd, XA_API_CMD_INIT, XA_CMD_TYPE_INIT_DONE_QUERY,
+		 &codec->cpd.init_done, ret);
```

Comment:
side note unrelated to this PR: I think it would be good to switch `API_CALL()` to
```
#define API_CALL(cd, cmd, sub_cmd, value) \
		(cd)->api((cd)->self, cmd, sub_cmd, value)
```
and then do explicitly
```
ret = API_CALL(cd, cmd, sub_cmd, value);
```

Reply:
@lyakh not a bad idea, but I don't see a draw back from the current implementation either. It is just a mater of taste. We all know that API_CALL is a macro. 

Remember `spin_lock_irqsave` in the Linux kernel?